src/mds/MDCache.h

```diff
@@ -68,10 +69,6 @@ class MMDSFragmentNotify;
 
 class ESubtreeMap;
 
-struct MDRequestImpl;
-typedef ceph::shared_ptr<MDRequestImpl> MDRequestRef;
-struct MDSlaveUpdate;
```

Comment:
Oh, and we imply use of all those functions by using the intrusive_ptr as a parameter? *sigh*